### ISSUE TITLE
Update the API POST Users endpoint (`/api/users`) so that it correctly creates a new User model

### ISSUE THREAD

**Issue by abrahmasandra** [uchicago-cs/chigame#218]
Currently, the API `POST` users endpoint at `/api/users/` treats the `User` model as a normal model, i.e. POSTing using a normal JSON post request (through Django REST Framework). This will actually not work to create new users because the `User` model has special attributes, specifically the `password` attribute. We will need overwrite the `create()` method of the `UserListView` in order to create a `User` object using the python command built into Django. 

We will know this issue is completed when the `POST` request to the users endpoint `/api/users` creates a new `User` model in our database. This will be reflected if we can see it when we make a `GET` request to the users endpoint.

**Comment by abrahmasandra** [comment]
`POST` requests to the `/api/users/` endpoint now work (i.e. they create a new User with a given email and password). I have manually tested using the API to create users and then logging in through the website. 

The `UserSerializer` now uses the built in `create_user()` method (from the UserManager class) when creating an object from the JSON data. This way of creating users allows for the users to actually be stored in the database.

**Comment by abrahmasandra** [comment]
I also had to add a `username` attribute to the `User` model. We want to do this, so a user can specify a given username when logging in or signing up. We also wanted to do this for the future, because we hope to implement slugs that use a user's `username` to get the Detail View information of the `User`

**Comment by abrahmasandra** [comment]
Closing Comment: Successfully, updated the POST Users endpoint at `/api/users/` so that it correctly creates a new `User` model in the database. In order to do this, I needed to override the default method that the `UserSerializer` used for creating a new object, and instead needed to use the `create_user` method within the `UserManager` class.

Now, we can create new `User` objects along with inputting `passwords` through POST requests. In addition, a `username` attribute was added to the `User` class in order to act as `slugs` in other URLs
